src/services/worker/tasks/fetch.task.ts

```diff
@@ -45,45 +42,62 @@ export class FetchTask {
       );
     }
 
+    const latestSyncedBlockNumber = await this._systemService.get(
+      SystemType.GENERAL_LAST_BLOCK_NUMBER,
+    );
+
+    const latestBlockNumber =
+      await this._web3Provider.web3.eth.getBlockNumber();
+
+    if (
+      latestSyncedBlockNumber + FetchTask.MAX_DELAY_BLOCK_RANGE >
+      latestBlockNumber
+    ) {
+      // this._logger.log(`Ignore task. Allowed delay range.`);
+      return;
+    }
+
+    FetchTask.isProcessLocked = true;
+
     try {
-      FetchTask.isProcessLocked = true;
+      const fromBlock =
+        latestSyncedBlockNumber || this._web3Provider.getGenesisBlock();
 
-      while (true) {
-        const fromBlock =
-          (await this._systemService.get(
-            SystemType.GENERAL_LAST_BLOCK_NUMBER,
-          )) || this._web3Provider.getGenesisBlock();
+      const toBlock = Math.min(
+        fromBlock + FetchTask.BLOCK_RANGE,
+        latestBlockNumber,
+      );
 
-        const toBlock = Math.min(
-          fromBlock + FetchTask.BLOCK_RANGE,
-          latestBlockNumber,
-        );
+      this._logger.log(
+        `Syncing events in a blocks range: ${fromBlock} - ${toBlock}`,
+      );
 
-        this._logger.log(
-          `Syncing events in a blocks range: ${fromBlock} - ${toBlock}`,
-        );
+      await this._syncUpdates(fromBlock, toBlock);
 
-        await this._syncUpdates(fromBlock, toBlock);
+      this._logger.log(
+        `Sync completed for a blocks range: ${fromBlock} - ${toBlock}`,
+      );
 
+      // Metrics
+      this._metricsService.fetchStatus.set(1);
+
+      const messageDetails = `From block: ${fromBlock}, to block: ${toBlock}, latest block: ${latestBlockNumber}`;
+      this._logger.log(
+        `Fetched all events till latest block. ${messageDetails}`,
+      );
+      /*
+      if (toBlock < latestBlockNumber) {
         this._logger.log(
-          `Sync completed for a blocks range: ${fromBlock} - ${toBlock}`,
+          `Continuing fetching immediately because latest block number is not reached. ${messageDetails}`,
```

Comment:
we need it?

Reply:
@vadiminc if logger supports debug messages and by default level is higher, let's keep it as debugs and uncomment